plugins/partiql-local/src/main/kotlin/org/partiql/plugins/local/LocalSchema.kt

```diff
@@ -149,6 +153,27 @@ public fun StructElement.toStructType(): StaticType {
     return StructType(fields, contentClosed, constraints = constraints)
 }
 
+public fun StructElement.toDecimalType(): StaticType {
+    val precision = get("precision").bigIntegerValue.intValueExact()
+    val scale = get("scale").bigIntegerValue.intValueExact()
+    val precisionScaleConstraint = DecimalType.PrecisionScaleConstraint.Constrained(precision, scale)
+    return DecimalType(precisionScaleConstraint)
+}
+
+public fun StructElement.toStringType(): StaticType {
```

Comment:
(minor) btw, for Scribe we had added constraints to the LocalSchema Ion reader -- https://github.com/partiql/partiql-scribe/pull/19. Could consider reusing that logic or making things the same in the future.

One difference is default values for decimal scale if unspecified. Also separating out string into string (i.e. `VARCHAR`) and char (i.e. `CHAR`).

Reply:
Thanks for the context. 

Perhaps we can do this in a follow up PR, the reason being split string and char causes potentially a lot diff in our tests.